frontend/src/ClientDataForm.js

```diff
@@ -116,19 +122,57 @@ const ClientDataForm = () => {
     </div>
   );
 
+  if (loading) {
+    return (
+      <>
+        <div className="container d-flex justify-content-center align-items-center min-vh-100">
+          <div className="spinner-grow text-primary" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+          <div className="spinner-grow text-secondary" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+          <div className="spinner-grow text-success" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+          <div className="spinner-grow text-danger" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+          <div className="spinner-grow text-warning" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+          <div className="spinner-grow text-info" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+          <div className="spinner-grow text-dark" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+        </div>
+      </>
+    );
+  }
+
   return (
     <>
       <div className="container">
         <div className="row mb-2 border-bottom">
           <h1>Client Form</h1>
         </div>
         <form method="post" onSubmit={log_information}>
-          {questions.map((question, index) => renderQuestion(question, index))}
+          {questions
+            .filter((question) => question.language_fk_id == language)
+            .map((question, index) => renderQuestion(question, index))}
           <div className="row my-2">
             <div className="col mb-2" align="center">
-              <button className="btn btn-primary btn-lg" type="submit">
-                Submit
-              </button>
+              <Link to="/selectlanguage">
```

Comment:
Clicking submit doesn't currently add data to the database, needs to fix that before merging.